src/Raven.Server/Documents/Queries/GraphQueryOrderByFieldComparer.cs

```diff
@@ -0,0 +1,235 @@
+            {
+                case LazyStringValue xLazyStringValue:
+                    if (yObject is LazyStringValue yLazyStringValue)
+                    {
+                        return xLazyStringValue.CompareTo(yLazyStringValue) * _order;
+                    }
+                    else if (yResult is string yStringValue)
+                    {
+                        return xLazyStringValue.CompareTo(yStringValue) * _order;
+                    }
+                    return LogMissmatchTypesReturnOrder(xObject, yObject);
+                    break;
+                case string xString:
+                    if (yObject is string yString)
+                    {
+                        return string.CompareOrdinal(xString, yString) * _order;
+                    }
+                    return LogMissmatchTypesReturnOrder(xObject, yObject);
+                case LazyNumberValue xLazyNumberValue:
+                    switch (yObject)
+                    {
+                        case int i:
+                            return xLazyNumberValue.CompareTo(i) * _order;
+                        case long l:
+                            return xLazyNumberValue.CompareTo(l) * _order;
+                        case double d:
+                            return xLazyNumberValue.CompareTo(d) * _order;
+                        case float f:
+                            return xLazyNumberValue.CompareTo(f) * _order;
+                        case LazyNumberValue lnv:
+                            return xLazyNumberValue.CompareTo(lnv) * _order;
+                        default:
+                            return LogMissmatchTypesReturnOrder(xObject, yObject);
+                    }
+                case long xLong:
+                    if (yObject is long yLong)
+                    {
+                        return xLong.CompareTo(yLong) * _order;
+                    }
+                    return LogMissmatchTypesReturnOrder(xObject, yObject);
+                case IComparable xComparable:
+                    if(yObject.GetType() == xObject.GetType())
+                        return xComparable.CompareTo(yObject) * _order;
+                    return LogMissmatchTypesReturnOrder(xObject, yObject);
+                case BlittableJsonReaderArray xBlittableJsonReaderArray:
+                    if (yObject is BlittableJsonReaderArray yBlittableJsonReaderArray)
+                    {
+                        return string.CompareOrdinal(xBlittableJsonReaderArray.ToString(), yBlittableJsonReaderArray.ToString());
+                    }
+                    return LogMissmatchTypesReturnOrder(xObject, yObject);
+                case BlittableJsonReaderObject xBlittableJsonReaderObject:
+                    if (yObject is BlittableJsonReaderObject yBlittableJsonReaderObject)
+                    {
+                        return string.CompareOrdinal(xBlittableJsonReaderObject.ToString(), yBlittableJsonReaderObject.ToString());
```

Comment:
There is no meaningful way to compare them, and I certainly don't want to use strings.
Use the `Base` as a way to get just a stable sort order.